interface/src/app/greenelab/utils.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * This module includes a few utility components that will be shared in other
+ * modules to avoid duplicate codes.
+ */
+
+angular.module('greenelab.utils', [])
+.constant('ApiBasePath', '/api/v0/')
+
+// A service function that returns a string of error message based on input
+// context and response.
+.value('errGen', function(context, response) {
```

Comment:
It is just a string that indicates the scenario. You can see an example in `annotation.js`.